lib/razor/cli/navigate.rb

```diff
@@ -5,14 +5,39 @@
 
 module Razor::CLI
   class Navigate
+    PRIORITY_RAZOR_API = "https://localhost:8151/api"
+    DEFAULT_RAZOR_API = "http://localhost:8150/api"
+    def self.default_api
+      DEFAULT_RAZOR_API
+    end
     extend Forwardable
 
     def initialize(parse, segments)
       @parse = parse
       @segments = segments||[]
+      set_api!(parse)
       @doc = entrypoint
       @doc_resource = create_resource parse.api_url, {:accept => :json,
-                                                      :accept_language => accept_language}
+                                               :accept_language => accept_language}
+    end
+
+    # The order of API selection works as follows:
+    # - Use `-u` argument if defined (done elsewhere)
+    # - Use "RAZOR_API" environment variable if defined
+    # - Check PE's https://localhost:8151/api via `HEAD` HTTP method
+    # - Use FOSS' http://localhost:8150/api
+    # This receives an argument determining whether to be verbose about
+    # requests made.
+    def set_api!(parse)
+      if !!parse.api_url
+        parse.api_url.to_s
+      elsif ENV["RAZOR_API"]
+        parse.parse_and_set_api_url(ENV['RAZOR_API'], :env)
+      elsif check_api
```

Comment:
I would rename this to `pe_razor_api_exists?` since that's what it's checking for (or something along those lines).

Reply:
yeah, I'm also into being super explicit with method names, so +1

Reply:
I'll make this more descriptive and use `check_priority_api_exists?`, keeping with a preference for "priority" over "PE", explained above.